app/routes/synonyms.$synonymId._index.tsx

```diff
@@ -0,0 +1,176 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import type { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node'
+import { redirect } from '@remix-run/node'
+import { Form, Link, useLoaderData } from '@remix-run/react'
+import { Button, ButtonGroup, FormGroup, Icon } from '@trussworks/react-uswds'
+import { useState } from 'react'
+
+import {
+  deleteSynonyms,
+  getSynonymsByUuid,
+  putSynonyms,
+} from './synonyms/synonyms.server'
+import { getFormDataString } from '~/lib/utils'
+
+export async function loader({ params: { synonymId } }: LoaderFunctionArgs) {
+  if (!synonymId) return { synonymId: null, eventIds: null }
+  const synonyms = await getSynonymsByUuid(synonymId)
+  const eventIds = [] as string[]
+  synonyms.forEach((synonym) => {
+    eventIds.push(synonym.eventId)
+  })
+
+  return {
+    synonymId,
+    eventIds,
+  }
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  const data = await request.formData()
+  const intent = getFormDataString(data, 'intent')
+  const synonymId = getFormDataString(data, 'synonymId')
+  if (!synonymId) return redirect('/synonyms')
```

Comment:
I don't think you need to get this from the form, because it's available in the route params.